src/libraries/System.Runtime/tests/System/DateTimeOffsetTests.cs

```diff
@@ -856,14 +856,18 @@ public static void Compare(DateTimeOffset dateTimeOffset1, DateTimeOffset dateTi
             }
         }
 
-        [Fact]
-        public static void Parse_String()
+        public static IEnumerable<object[]> Parse_TestData()
         {
-            DateTimeOffset expected = DateTimeOffset.MaxValue;
-            string expectedString = expected.ToString();
+            yield return new object[] { "2021-04-23T13:04:17,307642270+02:00", new DateTimeOffset(637547798573076423, new TimeSpan(2, 0, 0)) };
```

Comment:
@ericstj (or somebody else from area-System.DateTime), pay attention on this line, please. I think, expected value should be represented by [DateTimeOffset(Int32, Int32, Int32, Int32, Int32, Int32, Int32, TimeSpan)](https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset.-ctor?view=net-7.0#system-datetimeoffset-ctor(system-int32-system-int32-system-int32-system-int32-system-int32-system-int32-system-int32-system-timespan)), but max value of millisecond is 999, that's why I could not use this constructor

Reply:
You can do it like:

```C$

{ "2021-04-23T13:04:17,307000000+02:00", 
new DateTimeOffset(2021, 4, 23, 13, 4, 17, 307, TimeSpan.FromHours(2))

```

Reply:
@tarekgh , thanks, but next code result is false:
```
DateTimeOffset dt = new DateTimeOffset(2021, 4, 23, 13, 4, 17, 307, 642, TimeSpan.FromHours(2));
DateTimeOffset dt2 = DateTimeOffset.Parse("2021-04-23T13:04:17.307642270+02:00");
Console.WriteLine(dt == dt2);
```
because dt.Ticks is 63754779857307642**0** and dt2.Ticks is 63754779857307642**3** (last digit is different), that's why I had post my previous comment

Reply:
@Maximys any news?

Reply:
@tarekgh , yes, you are right, with 6 significant digits (I mean "307642" without "270") test will be successful, but inside bug we have "2021-04-23T13:04:17,307642270+02:00". As you can see, count of significant digits of "2021-04-23T13:04:17,307642270+02:00" is 8 ("30764227"), which add 3 to last digit of Ticks ("0.xxxx227" rounded to "0.xxxx3") and I think, that we should not use good data to test case just for pass it.
If you want to know my opinion, I think we should change type of "microsecond" argument from [Int32](https://learn.microsoft.com/en-us/dotnet/api/system.int32?view=net-7.0) to [float](https://learn.microsoft.com/en-us/dotnet/api/system.single?view=net-7.0) and apply range  0<=microsecond<1000. Then we can pass 642.3 to it and my test will be pass.

Reply:
> I think we should change type of "microsecond" argument from [Int32](https://learn.microsoft.com/en-us/dotnet/api/system.int32?view=net-7.0) to [float](https://learn.microsoft.com/en-us/dotnet/api/system.single?view=net-7.0)

we cannot do that for compatibility reasons. We never change APIs signature after we ship.